scripts/asm-installer/install_asm

```diff
@@ -1803,46 +1804,70 @@ EOF
 required_iam_roles() {
   # meshconfig.admin - required for init, stackdriver, UI elements, etc.
   # servicemanagement.admin/serviceusage.serviceUsageAdmin - enables APIs
-  cat <<EOF
-roles/servicemanagement.admin
-roles/serviceusage.serviceUsageAdmin
-roles/meshconfig.admin
-roles/compute.admin
-roles/container.admin
-roles/resourcemanager.projectIamAdmin
-roles/iam.serviceAccountAdmin
-roles/iam.serviceAccountKeyAdmin
-roles/gkehub.admin
-roles/privateca.admin
-EOF
+  if can_modify_gcp_components || \
+     can_modify_cluster_labels || \
+     can_modify_cluster_roles; then
+    echo roles/container.admin
+  fi
+  if can_modify_gcp_components; then
+    echo roles/meshconfig.admin
+  fi
+  if can_modify_gcp_apis; then
+    echo roles/servicemanagement.admin
+    echo roles/serviceusage.serviceUsageAdmin
+  fi
+  if can_modify_gcp_iam_roles; then
+    echo roles/resourcemanager.projectIamAdmin
+  fi
+  if is_sa; then
+    echo roles/iam.serviceAccountAdmin
+  fi
+  if can_register_cluster; then
+    echo roles/gkehub.admin
+  fi
+  if [[ "${CA}" = "gcp_cas" ]]; then
+    echo roles/privateca.admin
+  fi
+  if [[ "${_CI_I_AM_A_TEST_ROBOT}" -eq 1 ]]; then
+    echo roles/compute.admin
+    echo roles/iam.serviceAccountKeyAdmin
+  fi
 }
 # [END required_iam_roles]
 
 # [START required_apis]
 required_apis() {
     cat << EOF
-container.googleapis.com
-compute.googleapis.com
 monitoring.googleapis.com
 logging.googleapis.com
 cloudtrace.googleapis.com
 meshtelemetry.googleapis.com
-meshconfig.googleapis.com
-iamcredentials.googleapis.com
-gkeconnect.googleapis.com
-gkehub.googleapis.com
-cloudresourcemanager.googleapis.com
 stackdriver.googleapis.com
 EOF
-  case "${CA}" in
-   mesh_ca)
-     echo meshca.googleapis.com
-     ;;
-   gcp_cas)
-     echo privateca.googleapis.com
-     ;;
-    *);;
-  esac
+  if can_modify_gcp_components || \
+     can_modify_cluster_labels || \
+     can_modify_cluster_roles; then
+    echo container.googleapis.com
+  fi
+  if can_modify_gcp_components; then
+    echo meshconfig.googleapis.com
+  fi
+  if can_modify_gcp_iam_roles; then
+    echo cloudresourcemanager.googleapis.com
+    echo iamcredentials.googleapis.com
+  fi
+  if can_register_cluster; then
+    echo gkeconnect.googleapis.com
+    echo gkehub.googleapis.com
+  fi
```

Comment:
These APIs are required all of the time.

Reply:
Updated.